templates/core/terraform/state-store/statestore.tf

```diff
@@ -25,3 +25,34 @@ resource "azurerm_cosmosdb_sql_database" "tre-db" {
   account_name        = azurerm_cosmosdb_account.tre-db-account.name
   throughput          = 400
 }
+
+resource "azurerm_private_dns_zone" "statestore" {
+  name                = "privatelink.statestore.azure.net"
+  resource_group_name = var.resource_group_name
+}
+
+resource "azurerm_private_dns_zone_virtual_network_link" "statestorelink" {
```

Comment:
Again, can we name `cosmos_documents_dns_link` or similar s isn't state store specific.

Reply:
Same comment as above. I don't mind changing but I think statestore is more concrete than cosmos.